buckets: init at 0.75.0

kmogged opened the pull request:
Adds initial package for the financial applications buckets (https://www.budgetwithbuckets.com/)

I have updated the maintainers file with my info and have added myself as the maintainer. I use this software daily and have been using this particular nix definition for a few months now and I think the project is mature enough to be included in the nixpkgs. I have had the most success wrapping the AppImage. I am open to improvements, but the software is stable on NixOS.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

zimward commented on pkgs/by-name/bu/buckets/package.nix:
```diff
@@ -0,0 +1,66 @@
+in
+appimageTools.wrapType2 {
+  inherit pname version src;
+  name = "${pname}";
```
```suggestion
  name = pname;
```

zimward commented on pkgs/by-name/bu/buckets/package.nix:
```diff
@@ -0,0 +1,66 @@
+
+let
+
+  inherit (stdenv.hostPlatform) system;
+  throwSystem = throw "Unsupported system: ${system}";
+
+  platform =
+    {
+      x86_64-linux = "amd64";
+      aarch64-linux = "arm64";
+    }
+    .${system} or throwSystem;
```
this should probably be done with the `meta.broken` and/or `meta.platforms` attribute to ensure the it not being called with those systems accidentally instead of throwing.

zimward commented on pkgs/by-name/bu/buckets/package.nix:
```diff
@@ -0,0 +1,66 @@
+
+  pname = "buckets";
+  version = "0.75.0";
+  nameExecutable = pname;
```
`nameExecuteable` is not used.

zimward reviewed (commented):
Please squash the nixfmt commit. Adding yourself as a maintainer should be in a separate commit. See: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

zimward commented on pkgs/by-name/bu/buckets/package.nix:
```diff
@@ -0,0 +1,66 @@
+    done
+  '';
+
+  meta = with lib; {
+    description = "Private family budgeting app";
+    homepage = "https://www.budgetwithbuckets.com/";
+    license = licenses.unfree;
+    maintainers = with maintainers; [ kmogged ];
+    platforms = [
+      "x86_64-linux"
+      "aarch64-linux"
+    ];
```
```suggestion
  meta = {
    description = "Private family budgeting app";
    homepage = "https://www.budgetwithbuckets.com/";
    license = lib.licenses.unfree;
    maintainers = with lib.maintainers; [ kmogged ];
    platforms = [
      "x86_64-linux"
      "aarch64-linux"
    ];
```
See: #208242

kmogged committed:
maintainers: add kmogged

kmogged committed:
buckets: init at 0.75.0
Adds initial package for the financial application buckets (https://www.budgetwithbuckets.com/)

zimward commented on pkgs/by-name/bu/buckets/package.nix:
```diff
@@ -0,0 +1,65 @@
+in
+appimageTools.wrapType2 {
+  inherit pname version src;
+  name = pname;
+  extraPkgs = pkgs: (appimageTools.defaultFhsEnvArgs.multiPkgs pkgs) ++ [ pkgs.libsodium ];
```
```suggestion
  name = pname;
  
  extraPkgs = pkgs: (appimageTools.defaultFhsEnvArgs.multiPkgs pkgs) ++ [ pkgs.libsodium ];
```

zimward commented on pkgs/by-name/bu/buckets/package.nix:
```diff
@@ -0,0 +1,65 @@
+in
+appimageTools.wrapType2 {
+  inherit pname version src;
+  name = pname;
```
```suggestion
```
I as far as i can tell `wrapType2` does not expect a `name` attr.

kmogged replied:
Thank you for all the help. I have removed this and tested on 24.11.

zimward reviewed (approved):
LGTM apart from my nits.

FliegendeWurst commented on pkgs/by-name/bu/buckets/package.nix:
```diff
@@ -0,0 +1,65 @@
+      "aarch64-linux"
+    ];
+    mainProgram = "buckets";
+    broken = !stdenv.hostPlatform.isLinux;
```
```suggestion
```
Not needed with the platforms above.

kmogged replied:
Removed and makes total sense to me.

FliegendeWurst commented on pkgs/by-name/bu/buckets/package.nix:
```diff
@@ -0,0 +1,65 @@
+  extraPkgs = pkgs: (appimageTools.defaultFhsEnvArgs.multiPkgs pkgs) ++ [ pkgs.libsodium ];
+  extraInstallCommands = ''
+    source "${makeWrapper}/nix-support/setup-hook"
+    wrapProgram $out/bin/${pname}
```
```suggestion
    wrapProgram $out/bin/buckets
```
Same below.
#277994

kmogged replied:
Thank you for the insight. A lot of my nix configurations are reused, but scalability in nixpkgs is more important.

kmogged commented:
> Please rebase instead of merging.

Opps, I see what I did there, sorry. My branch now only has the 2 commits we care about.
